modules/channel-web/src/backend/socket.ts

```diff
@@ -25,46 +19,30 @@ export default async (bp: typeof sdk, db: Database) => {
       return next()
     }
 
+    const messaging = await db.getMessagingClient(event.botId)
     const messageType = event.type === 'default' ? 'text' : event.type
     const userId = event.target
-    const conversationId = event.threadId || (await db.getOrCreateRecentConversation(event.botId, userId))
-
-    if (!_.includes(outgoingTypes, messageType)) {
-      bp.logger.warn(`Unsupported event type: ${event.type}`)
-      return next(undefined, true)
-    }
-
-    const standardTypes = ['text', 'carousel', 'custom', 'file', 'login_prompt', 'video', 'audio']
+    const { visitorId } = await db.getMappingFromUser(userId)
+    const conversationId = event.threadId || (await messaging.conversations.getRecent(userId)).id
 
     if (!event.payload.type) {
       event.payload.type = messageType
     }
 
-    if (messageType === 'typing') {
+    if (messageType === 'data') {
+      const payload = bp.RealTimePayload.forVisitor(visitorId, 'webchat.data', event.payload)
+      bp.realtime.sendPayload(payload)
+    } else {
       const typing = parseTyping(event.payload.value)
-      const payload = bp.RealTimePayload.forVisitor(userId, 'webchat.typing', { timeInMs: typing, conversationId })
+      const payload = bp.RealTimePayload.forVisitor(visitorId, 'webchat.typing', { timeInMs: typing, conversationId })
       // Don't store "typing" in DB
       bp.realtime.sendPayload(payload)
```

Comment:
There is a property called "typing" in the payloads which determines if we send the typing event or not. Here typings are sent all the time